packages/zero-solid/src/create-query.ts

```diff
@@ -38,25 +38,25 @@ export function createQuery<
   querySignal: () => Query<TSchema, TTable, TReturn>,
   options?: CreateQueryOptions | Accessor<CreateQueryOptions>,
 ): QueryResult<TReturn> {
-  // Wrap in in createMemo to ensure a new view is created if the querySignal changes.
-  const view = createMemo(() => {
+  const [state, setState] = createStore<State>([
```

Comment:
Just to be clear. The old implementation cached the store between different calls to createQuery and shared as long as the query object was the same.

I think this new semantics is fine but I wanted to know the main motivation for this change.

Reply:
The Query object should change on every dependency change since queries are immutable.

The question if the store should change is orthogonal. I do buy the argument that the store should not change because that allows the UI to do less updates.